components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -75,16 +82,29 @@ void ViewCounterModel::RegisterPageViewForBrandedImages() {
   // view of the branded wallpaper.
   count_to_branded_wallpaper_--;
   if (count_to_branded_wallpaper_ < 0) {
-    // Reset count and increse image index for next time.
+    // Reset count and randomize image index for next time.
     count_to_branded_wallpaper_ = kRegularCountToBrandedWallpaper;
-    campaigns_current_branded_image_index_[current_campaign_index_]++;
-    campaigns_current_branded_image_index_[current_campaign_index_] %=
-        campaigns_total_branded_image_count_[current_campaign_index_];
 
-    // Increse campaign index for next time.
-    current_campaign_index_++;
-    current_campaign_index_ %= total_campaign_count_;
-    return;
+    if (always_show_branded_wallpaper_) {
```

Comment:
How about introducing member vars such as `shuffle_` or similar to `ViewCounterModel` instead of checking `always_show_branded_wallpaper_`?
and then `SetCampaignsTotalBrandedImageCount()` and `RegisterPageViewForBrandedImages()` calls different logic based on that flag.
Test case also could test both mode with that flag.

Reply:
Discussed in DM and will follow-up with a PR to decouple SI and SR